File: proyecto-backend-nexu/src/main/java/com/nexu/PlacesToEat/dto/PlacesToEatRequestDto.java
package com.nexu.PlacesToEat.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.nexu.Places.domain.enums.Payment;
import com.nexu.PlacesToEat.domain.enums.PlaceCategoryToEat;
import com.nexu.PlacesToEat.domain.enums.TypeCoffee;
import com.nexu.PlacesToEat.domain.enums.TypeRestaurant;
import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
public class PlacesToEatRequestDto {

    @NotBlank(message = "El nombre es obligatorio")
    private String name;

    @NotBlank(message = "La dirección es obligatoria")
    private String address;

    @NotNull
    private Double latitude;

    @NotNull
    private Double longitude;

    @NotNull
    private Payment payment;

    @NotNull
    @JsonFormat(pattern = "HH:mm")
    private LocalTime openTime;

    @NotNull
    @JsonFormat(pattern = "HH:mm")
    private LocalTime closeTime;

    @NotBlank
    private String description;

    private Boolean wifi;

    private String priceRange;

    private Double estimatedPrice;

    @NotNull
    private Integer capacity;

    @NotBlank
    private String status;

    // específicos de PlacesToEat
    @NotNull
    private PlaceCategoryToEat placeCategoryToEat;
    private TypeRestaurant typeRestaurant;
    private Boolean delivery;
    private TypeCoffee typeCoffee;
    private String menu;
}

File: proyecto-frontend-mobile-nexu3/src/interfaces/placesToEat/PlacesToEatRequestDto.ts
export interface PlacesToEatRequestDto {
  name: string;
  address: string;
  latitude: number;
  longitude: number;
  payment: Payment;
  openTime: string; // HH:mm
  closeTime: string; // HH:mm
  description: string;
  wifi?: boolean;
  priceRange?: string;
  estimatedPrice?: number;
  capacity: number;
  status: string;
  placeCategoryToEat: PlaceCategoryToEat;
  typeRestaurant?: TypeRestaurant;
  delivery?: boolean;
  typeCoffee?: TypeCoffee;
  menu?: string;
}

export type Payment = 'FREE' | 'CARD' | 'CASH' | 'YAPE';
export type PlaceCategoryToEat = 'COFFEE' | 'RESTAURANT';
export type TypeRestaurant = 'CRIOLLO' | 'SELVATICO' | 'MARISCO';
export type TypeCoffee = 'TRADICIONAL' | 'PETFRIENDLY' | 'VEGANA'; 